In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
import pandas as pd
import os
import shutil

In [3]:
# create the data for postive samples (Github)

FILE_PATH = '/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Covid-19_Chest_X-ray_Dataset/metadata.csv'
IMAGES_PATH = '/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Covid-19_Chest_X-ray_Dataset/images'

In [4]:
df = pd.read_csv(FILE_PATH)
print(df.shape)

(950, 30)


In [5]:
df.head(11)

patientid  offset  ... other_notes  Unnamed: 29
0          2     0.0  ...         NaN          NaN
1          2     3.0  ...         NaN          NaN
2          2     5.0  ...         NaN          NaN
3          2     6.0  ...         NaN          NaN
4          4     0.0  ...         NaN          NaN
5          4     5.0  ...         NaN          NaN
6          5     NaN  ...         NaN          NaN
7          6     0.0  ...         NaN          NaN
8          6     4.0  ...         NaN          NaN
9          3     4.0  ...         NaN          NaN
10         3     9.0  ...         NaN          NaN

[11 rows x 30 columns]

In [6]:
TARGET_DIR = "/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Dataset/Covid-19"

if not os.path.exists(TARGET_DIR):
  os.mkdir(TARGET_DIR)
  print("Covid-19 folder created")

Covid folder created


In [7]:
cnt = 0

for (i,row) in df.iterrows():
  if row["finding"] == "Pneumonia/Viral/COVID-19"  and (row['view'] == 'PA' or row['view'] == 'AP'):
    filename = row["filename"]
    image_path = os.path.join(IMAGES_PATH,filename)
    image_copy_path = os.path.join(TARGET_DIR,filename)
    shutil.copy2(image_path, image_copy_path)
    print("Moving image ", cnt)
    cnt = cnt + 1


Moving image  0
Moving image  1
Moving image  2
Moving image  3
Moving image  4
Moving image  5
Moving image  6
Moving image  7
Moving image  8
Moving image  9
Moving image  10
Moving image  11
Moving image  12
Moving image  13
Moving image  14
Moving image  15
Moving image  16
Moving image  17
Moving image  18
Moving image  19
Moving image  20
Moving image  21
Moving image  22
Moving image  23
Moving image  24
Moving image  25
Moving image  26
Moving image  27
Moving image  28
Moving image  29
Moving image  30
Moving image  31
Moving image  32
Moving image  33
Moving image  34
Moving image  35
Moving image  36
Moving image  37
Moving image  38
Moving image  39
Moving image  40
Moving image  41
Moving image  42
Moving image  43
Moving image  44
Moving image  45
Moving image  46
Moving image  47
Moving image  48
Moving image  49
Moving image  50
Moving image  51
Moving image  52
Moving image  53
Moving image  54
Moving image  55
Moving image  56
Moving image  57
Moving image  58
Moving 

In [8]:
print("Total number of covid-19 images collected --> ", cnt)

Total number of covid-19 images collected -->  342


total covid positive images --> 584

PA View --> 196 images

AP View --> 146 images

total collected PA and AP view images --> 342

now we will do the same for normal and pneumonia images.

sampling of images from kaggle dataset

In [2]:
import random

KAGGLE_FILE_PATH_NORMAL = "/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Chext_X-ray_Kaggle/train/NORMAL"
KAGGLE_FILE_PATH_PNEUMONIA = "/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Chext_X-ray_Kaggle/train/PNEUMONIA"

TARGET_FILE_PATH_NORMAL = "/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Dataset/Normal"
TARGET_FILE_PATH_PNEUMONIA = "/content/drive/My Drive/COVID-19_Project/Dataset_Processing/Dataset/Pneumonia"


since we will be increasing the covid images 2-4 times by using data augmentation resulting 684-1368 images, we will be sampling near to 1368 images from the repective folders also.

sampling images from normal 1340

sampling images from pneumonia 1370

In [5]:
image_names_normal = os.listdir(KAGGLE_FILE_PATH_NORMAL)
image_names_pneumonia = os.listdir(KAGGLE_FILE_PATH_PNEUMONIA)

In [6]:
random.shuffle(image_names_normal)
random.shuffle(image_names_pneumonia)

In [7]:
len(image_names_normal)

1341

In [8]:
len(image_names_pneumonia)

3875

for normal images

In [9]:
cnt = 0

for i in range (1340):

  image_name = image_names_normal[i]
  image_path = os.path.join(KAGGLE_FILE_PATH_NORMAL, image_name)

  target_path = os.path.join(TARGET_FILE_PATH_NORMAL, image_name)

  shutil.copy2(image_path, target_path)

  print("Moving image ", cnt)

  cnt = cnt + 1


Moving image  0
Moving image  1
Moving image  2
Moving image  3
Moving image  4
Moving image  5
Moving image  6
Moving image  7
Moving image  8
Moving image  9
Moving image  10
Moving image  11
Moving image  12
Moving image  13
Moving image  14
Moving image  15
Moving image  16
Moving image  17
Moving image  18
Moving image  19
Moving image  20
Moving image  21
Moving image  22
Moving image  23
Moving image  24
Moving image  25
Moving image  26
Moving image  27
Moving image  28
Moving image  29
Moving image  30
Moving image  31
Moving image  32
Moving image  33
Moving image  34
Moving image  35
Moving image  36
Moving image  37
Moving image  38
Moving image  39
Moving image  40
Moving image  41
Moving image  42
Moving image  43
Moving image  44
Moving image  45
Moving image  46
Moving image  47
Moving image  48
Moving image  49
Moving image  50
Moving image  51
Moving image  52
Moving image  53
Moving image  54
Moving image  55
Moving image  56
Moving image  57
Moving image  58
Moving 

for pneumonia images

In [10]:
cnt = 0

for i in range (1370):

  image_name = image_names_pneumonia[i]
  image_path = os.path.join(KAGGLE_FILE_PATH_PNEUMONIA, image_name)

  target_path = os.path.join(TARGET_FILE_PATH_PNEUMONIA, image_name)

  shutil.copy2(image_path, target_path)

  print("Moving image ", cnt)

  cnt = cnt + 1


Moving image  0
Moving image  1
Moving image  2
Moving image  3
Moving image  4
Moving image  5
Moving image  6
Moving image  7
Moving image  8
Moving image  9
Moving image  10
Moving image  11
Moving image  12
Moving image  13
Moving image  14
Moving image  15
Moving image  16
Moving image  17
Moving image  18
Moving image  19
Moving image  20
Moving image  21
Moving image  22
Moving image  23
Moving image  24
Moving image  25
Moving image  26
Moving image  27
Moving image  28
Moving image  29
Moving image  30
Moving image  31
Moving image  32
Moving image  33
Moving image  34
Moving image  35
Moving image  36
Moving image  37
Moving image  38
Moving image  39
Moving image  40
Moving image  41
Moving image  42
Moving image  43
Moving image  44
Moving image  45
Moving image  46
Moving image  47
Moving image  48
Moving image  49
Moving image  50
Moving image  51
Moving image  52
Moving image  53
Moving image  54
Moving image  55
Moving image  56
Moving image  57
Moving image  58
Moving 

total images collected -->

Covid-19: 342

Normal: 1340

Pneumonia: 1370